src/Pos/Explorer/Web/Server.hs

```diff
@@ -250,6 +298,47 @@ getTxSummary cTxId = do
 -- Helpers
 --------------------------------------------------------------------------------
 
+-- | Find all `MainBlock` by applying the *predicate*, starting from *headerHash*
+filterMainBlocks
+    :: (SscHelpersClass ssc, ExplorerMode m)
+    => HeaderHash
+    -> (MainBlock ssc -> Bool)
+    -> m [MainBlock ssc]
+filterMainBlocks headerHash predicate = rights <$> generalBlockSearch
+  where
+    generalBlockSearch    = filterAllBlocks headerHash specializedPred (pure [])
+    specializedPred block = either (const False) predicate block
```

Comment:
I can change that, but I did mean `specialized`, as in "this predicate is **specialized** for the `MainBlock`".